update-comments-and-fix-bag

GeorgiiGoshan opened the pull request:
1. Добавил комментарии, теперь программа расписана максимально подробно.
2. Исправил баг при котором во время повторного расчета не менялись значения некоторых переменных.

GeorgiiGoshan committed:
update-comments-and-fix-bag

setfed commented on plita.py:
```diff
@@ -222,12 +230,14 @@ def show_values(): # выводим результаты на экран при
 F.grid(row=14,column=1)
 
+# Создаем словарь для значений
 entries = {"q": q, "a1": a1, "b1": b1, "zox": zox, "zoy": zoy, "N1": N1, "N2": N2, "q1": q1, "Asw": Asw, "Sw": Sw, "Rs": Rs, "Rbt": Rbt, "h": h, "F": F}
```
Я бы предложил создать более осмысленные названия переменных
Ну типа а1 - это наверно ширина, так и можно написать "width" или "construction_plate_width"

GeorgiiGoshan replied:
Дело в том, что эти переменные, чтобы не запутать себя и пользователя программы называются точно так же, как они называются в методичке
При создании окна для ввода существует описание каждой конкретной переменной, например:
label_3 = tk.Label(win, text='Первый размер стороны поперечного сечения колонны a1, м: ')
Может еще сделать длинный комментарий перед созданием словаря для значений? С описанием этих значений

setfed replied:
Я понимаю, что эти все расчеты для внутреннего пользования и человек, который будет ими пользоваться наверно поймет
Только вот любой человек, который не разбирается в строительстве и впервые посмотрит на эти переменные, которые абсолютно похожи не поймет ничего 
![image](https://github.com/GeorgiiGoshan/calculation-of-the-plate-for-punching/assets/67071375/d1c4b4ea-fde1-4fa7-bfcf-2241655d2f7b)

Выяснение требований и полное понимание всех переменных и бизнес требований - один из навыков разработчика
То есть тебе так или иначе придется с этим работать

Если назвать переменные невозможно, можно просто описать комментарием каждую переменную

типа `zoy // диаметр колонны горизонтальный`

GeorgiiGoshan replied:
добавлены комментарии

GeorgiiGoshan replied:
Если по всем 3 замечаниям все устраивает сольем эту ветку с основной? чтобы для новых обновлений уже новая ветка была

setfed commented on plita.py:
```diff
@@ -95,60 +99,64 @@ def calculate(values): # считаем по формулам
 
-win = tk.Tk()
-##photo = tk.PhotoImage(file='fit.png')
+win = tk.Tk() # создаем окно приложения
```
Предлагаю создание окна программы выделить хотя бы отдельную функцию
Типа "def construct app window" 

 Внутри этой функции будут другие типа
 "adjust size" и подобное
 
Выделив функцию тебе легче будет вносить вносить изменения в дальнейшем

GeorgiiGoshan replied:
Создание окна и его настройка вынесены в отдельную функцию

setfed replied:
с 231 до 310 тоже вынести бы в отдельную функцию, это же тоже настройки

GeorgiiGoshan replied:
> с 231 до 310 тоже вынести бы в отдельную функцию, это же тоже настройки

пока даже с функцией не могу существенно сократить код, все равно приходится помимо самой функции прописывать каждый раз 
label_1 = create_label(win, 'Введите данные согласно указанным единицам: ', 0, 0)
q = tk.DoubleVar()

размер кода остается таким же, а на счет того, что становится удобнее читать не уверен. Или становится?

setfed replied:
Написание хорошего кода это не про количество строк, а про понятность, корректное разделение и пр.

Ну если не уверен - не выделяй

setfed commented on plita.py:
```diff
@@ -95,60 +99,64 @@ def calculate(values): # считаем по формулам
 ##win.iconphoto(False, photo)
-win.title('Расчёт плиты на продавливание')
+win.title('Расчёт плиты на продавливание') # задаем название вверху окна
 win.geometry('1100x600+250+50') # задаем размер и адрес окна
```
Это так называемые magic numbers

Предлагаю выделить в константы
Или пойти еще дальше, чтобы эти константы брались из некого конфигурационного файла

[вот](https://habr.com/ru/articles/485236/) что-то по этому поводу пишут 

В итоге предлагаю либо выделить константы, либо прописать в конфиг файлы

То есть не `win.geometry('1100x600+250+50')` а `win.geometry('WINDOW_HEIGHTxWINDOW_WIDTH')`

GeorgiiGoshan replied:
Пока мало опыта работы с конфиг файлами, поэтому вынес их в отдельные переменные (тоже в новой функции создания и работы с окном)

setfed replied:
Окей, конфиги в другой раз

GeorgiiGoshan committed:
1. Добавлены комментарии к переменным и формулам
2. Создана отдельная функция для создания окна приложения
3. Вынесены отдельно переменные размеров

setfed commented on plita.py:
```diff
@@ -1,55 +1,68 @@
-
-            if F_str == "N1+Fq1": # Считаем саму формулу
-                F_value = N1 + Fq1
+            Aq = h0 * (a1 + b1 + h0)        # площадь зоны продавливания расположенная вокруг колонны на расстоянии 1/2*h0 от контура поперечного сечения колонны
```
Переменные все равно выглядят так себе
Тем более, что твои комментарии раскрывают потенциал для переименования
Типа `area_of_column_punching` 

Так и быть, переименование сделаешь в другом PR

GeorgiiGoshan replied:
Сами переменные переименовывать точно не вариант, они должны быть точно такие же как в методичке, по которой пишется программа. Может сделать отдельный ридми в котором будут описаны все переменные и формулы?

setfed replied:
Кому они должны быть как в методичке?
Это же твой код, сделай как тебе удобно

А для клиента сделай какой-то маппер твоих переменных в инженерные a1, a2 и т.п.

GeorgiiGoshan replied:
> Кому они должны быть как в методичке? Это же твой код, сделай как тебе удобно
> 
> А для клиента сделай какой-то маппер твоих переменных в инженерные a1, a2 и т.п.

1. Для меня в первую очередь они должны быть как в методичке, потому что по ней пишется программа, и постоянно сверять разные названия одной и той же переменной не вариант, учитывая то, что методичка пройдена по факту только на 1/5, то есть по ней еще будет много работы.
2. Для клиента который будет видеть только само приложение? Все входные данные расписаны прямо в нём, он их видит. На счет выходных значений - желания такого у заказчика не поступало, приложение довольно профильное и лишний повторяющий то что он и так знает текст на экране ему не нужен, он и так знает какие переменные он ждет на выходе.

GeorgiiGoshan replied:
![image](https://github.com/GeorgiiGoshan/calculation-of-the-plate-for-punching/assets/139330449/d47dfa9d-1388-41b0-833a-b504a23f4407)

вот так это выглядит в самом приложении, все входные переменные подписаны

setfed commented on plita.py:
```diff
@@ -121,8 +134,10 @@ def calculate(values):  # считаем по формулам
+    # добавляем формулы для арматуры
+        Us = Ub
+        Rsw  = 0.8 * Rs
+        if Rsw > 300000:  # Rsw не может быть больше 300000мПа
```
Константы, константы, заведи парочку

типа `MAX_????_THRESHOLD`

setfed reviewed (commented):
Покомментил
Если требуется новые функциональные изменения - создавай другую ветку и новый PR 
никто тебя не остановит)
